src/main/scala/org/allenai/ike/BlackLabResult.scala

```diff
@@ -10,7 +9,8 @@ case class BlackLabResult(
     wordData: Seq[WordData],
     matchOffset: Interval,
     captureGroups: Map[String, Interval],
-    corpusName: String
+    corpusName: String,
+    sourceDocument: Option[Int] = None
```

Comment:
We should probably store a key we control, as opposed to a docId assigned by BlackLab.  Additionally, we can store an optional copy of the pre-tokenized text and extract that here:
```
  ...
  corpusName: String,
  sourceText: Option[String] = None,
  documentKey: Option[String] = None
```